lib/travis/build/addons/browserstack.rb

```diff
@@ -0,0 +1,153 @@
+require 'travis/build/addons/base'
+
+module Travis
+  module Build
+    class Addons
+      class BrowserStack < Base
+
+        BROWSERSTACK_HOME = '$HOME/.browserstack'
+        BROWSERSTACK_BIN_FILE = 'BrowserStackLocal'
+        BROWSERSTACK_BIN_URL = 'https://www.browserstack.com/browserstack-local'
+        ENV_KEY = 'BROWSERSTACK_KEY'
+        ENV_LOCAL = 'BROWSERSTACK_LOCAL'
+
+        def before_before_script
+          browserstack_key = access_key.to_s
+          sh.fold "browserstack.local.install" do
+            if browserstack_key.empty?
+              sh.echo "You provided an invalid access_key.", ansi: :red
+              return
+            end
+
+            sh.echo "Installing BrowserStack Local", ansi: :yellow
+            sh.if "$(uname) = 'Linux'" do
+              bin_package = "#{BROWSERSTACK_BIN_FILE}-linux-x64.zip"
+              bin_url = "#{BROWSERSTACK_BIN_URL}/#{bin_package}"
+
+              sh.cmd "mkdir -p #{BROWSERSTACK_HOME}"
+              sh.cmd "wget -O /tmp/#{bin_package} #{bin_url}", echo: true, timing: true, retry: true
+              sh.cmd "unzip -d #{BROWSERSTACK_HOME}/ /tmp/#{bin_package} 2>&1 > /dev/null", echo: false
+              sh.chmod "+x", "#{BROWSERSTACK_HOME}/#{BROWSERSTACK_BIN_FILE}", echo: false
+            end
+            sh.else do
+              sh.echo "Unsupported platform: $TRAVIS_OS_NAME.", ansi: :yellow
```

Comment:
Thanks for the careful review, @BanzaiMan. I couldn't get around to setting up a test environment on Mac OSX, so had left it for later. I've included it now. Should work just fine.